### PR TITLE
Added ability to process objects and arrays in operations

### LINKED ISSUE #15276: Engine - array & object direct mapping
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 5.0 | Engine | - | - | - |

We want to support object and array mapping with a more concise syntax. We want to go from:

```
- map:
    - key1.key2.key3. value
    - key1.key2.key4: value
    - key1.key2.key5: value
```

to:

```
- map:
  - key1:
    - key2:
      - key3: value
      - key4: value
      - key5. value
```

We will support both approaches, and the users will be able to use one or the other as they see fit.

## Tasks
- [x] Implement functionality
- [x] Update Design Document
- [x] Update script to generate a schema that supports new nested definitions
- [ ] QA validate

### PR DESCRIPTION
**Pull request by JavierBejMen**
|Related issue|
|---|
|Closes #15276 |

|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 5.0 | Engine | - | - | - |

Now, when mapping and checking conditions on objects and arrays, all nested operations are processed normally, handling references and function helpers properly. Before objects and arrays were handled as literal values.

This functionality is required to map the `threat` ECS fields without having to write the full path for each operation. E.g.:
```yml
map:
  - threat.enrichments:
      - indicator:
          description:  A local account created a new session 
          type: user-account
          reference: http://documentation.wazuh.com/… 
          provider: wazuh
          confidence: High
        matched:
          atomic: $user.name
          field: user.name
          type: indicator_match_rule
          occurred: +now//
          
      - indicator: ...
        matched: ...
```

### CODE REVIEW

File: src/engine/ruleset/schemas/ecs_types.json [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
   "geo_point",
-  "scaled_float",
   "keyword",
   "nested",
+  "scaled_float",
+  "text",
+  "boolean",
+  "date",
   "float"
 ]
```
Comment (JcabreraC) [inline src/engine/ruleset/schemas/ecs_types.json:13]: Inserts a line break at the end of the file.

**Review by JcabreraC — CHANGES_REQUESTED** [review]

File: src/engine/ruleset/schemas/ecs_types.json (hunk 1 above)
Comment (JavierBejMen) [reply src/engine/ruleset/schemas/ecs_types.json:13]: Done

**Review by LucioDonda — APPROVED** [review]
LGTM

**Review by LucioDonda — APPROVED** [review]

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Review by nmkoremblum — CHANGES_REQUESTED** [review]

**Pull request merged**